enterprise/frontend/src/metabase-enterprise/auth/components/SettingsSAMLForm/SettingsSAMLForm.tsx

```diff
@@ -0,0 +1,334 @@
+import CS from "metabase/css/core/index.css";
+import {
+  Form,
+  FormErrorMessage,
+  FormProvider,
+  FormSection,
+  FormSubmitButton,
+  FormTextInput,
+  FormTextarea,
+} from "metabase/forms";
+import { Box, Flex, Stack, Text, Title } from "metabase/ui";
+import { useUpdateSamlMutation } from "metabase-enterprise/api";
+import type { EnterpriseSettings } from "metabase-types/api";
+
+export type SAMLFormSettings = Pick<
+  EnterpriseSettings,
+  | "saml-user-provisioning-enabled?"
+  | "saml-attribute-email"
+  | "saml-attribute-firstname"
+  | "saml-attribute-lastname"
+  | "saml-identity-provider-uri"
+  | "saml-identity-provider-issuer"
+  | "saml-identity-provider-certificate"
+  | "saml-application-name"
+  | "saml-keystore-password"
+  | "saml-keystore-alias"
+  | "saml-keystore-path"
+  | "saml-attribute-group"
+  | "saml-group-sync"
+>;
+
+const SAML_FORM_SCHEMA = Yup.object({
+  "saml-attribute-group": Yup.string().nullable().default(null),
+});
+
+export function SettingsSAMLForm() {
+  const { data: settingDetails, isLoading: isLoadingDetails } =
+    useGetAdminSettingsDetailsQuery();
+  const { data: settingValues, isLoading: isLoadingValues } =
+    useGetSettingsQuery();
+  const [updateSamlSettings] = useUpdateSamlMutation();
+
+  const isEnabled = Boolean(settingValues?.["saml-enabled"]);
+
+  const handleSubmit = useCallback(
+    (values: SAMLFormSettings) => {
+      return updateSamlSettings({ ...values, "saml-enabled": true }).unwrap();
+    },
+    [updateSamlSettings],
+  );
+
+  // eslint-disable-next-line no-unconditional-metabase-links-render -- Admin settings
+  const { url: docsUrl } = useDocsUrl(
+    "people-and-groups/authenticating-with-saml",
+  );
+
+  const siteUrl = useSetting("site-url");
+
+  if (isLoadingDetails || isLoadingValues) {
+    return <LoadingAndErrorWrapper loading />;
+  }
+
+  if (!settingDetails || !settingValues) {
+    return (
+      <LoadingAndErrorWrapper error={t`Error loading SAML configuration`} />
+    );
+  }
+
+  return (
+    <Box maw="40rem" mx="md">
+      <FormProvider
+        initialValues={getFormValues(settingValues ?? {})}
+        onSubmit={handleSubmit}
+        validationSchema={SAML_FORM_SCHEMA}
+        enableReinitialize
+      >
+        {({ dirty }) => (
+          <Form>
+            <Breadcrumbs
+              className={CS.mb3}
+              crumbs={[
+                [t`Authentication`, "/admin/settings/authentication"],
+                [t`SAML`],
+              ]}
+            />
+            <Title order={2}>{t`Set up SAML-based SSO`}</Title>
+            <Text c="text-medium" mb="xl">
+              {jt`Use the settings below to configure your SSO via SAML. If you have any questions, check out our ${(
+                <ExternalLink
+                  key="link"
+                  href={docsUrl}
+                >{t`documentation`}</ExternalLink>
+              )}.`}
+            </Text>
+            {isEnabled && <SamlUserProvisioning />}
```

Comment:
Inside `SamlUserProvisioning` component it's also doing a check for
`const scimEnabled = useSetting("scim-enabled");`

Do we need to do this check twice?

Reply:
we're checking 2 different things. The form is checking if SAML is enabled, the user provisioning toggle is checking if scim is enabled.

One thing I want to revisit in MS3 is how we handle user provisioning  among scim, ldap, jwt and saml. it's confusing, both in the code, and I think for users, to have it all spread around like that